src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -436,8 +435,7 @@ def recover_vault(cmd, client, vault_name, resource_group_name, location, no_wai
     Sku = cmd.get_models('Sku', resource_type=ResourceType.MGMT_KEYVAULT)
     SkuName = cmd.get_models('SkuName', resource_type=ResourceType.MGMT_KEYVAULT)
     profile = Profile(cli_ctx=cmd.cli_ctx)
-    _, _, tenant_id = profile.get_login_credentials(
-        resource=cmd.cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    tenant_id = profile.get_subscription(subscription=cmd.cli_ctx.data['subscription_id'])[_TENANT_ID]
```

Comment:
This may result in `KeyError` as `subscription_id` may not always be in `cmd.cli_ctx.data`.

A little bit hacky, but `azure.cli.command_modules.role` accesses the private attribute `_config` of the client to retrieve the subscription ID which is from `cli_ctx.data['subscription_id']`:

https://github.com/Azure/azure-cli/blob/00fe3e551fbc574abfd3e099e8e15e6ecb623a06/src/azure-cli/azure/cli/command_modules/role/custom.py#L65-L66